src/main/java/com/example/weatherwaveapi/WeatherWaveApiApplication.java

```diff
@@ -1,9 +1,12 @@
 package com.example.weatherwaveapi;
 
+import com.example.weatherwaveapi.config.GeneralSettings;
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
 
 @SpringBootApplication
+@EnableConfigurationProperties(value = {GeneralSettings.class})
```

Comment:
1) Can be simplified to @EnableConfigurationProperties(GeneralSettings.class)
2) This annotation should be on the Configuration class